### PR TITLE
bpo-44339: Fix math.pow corner case to comply with IEEE 754

### PR DESCRIPTION
**Pull request by mdickinson**
This PR changes the behaviour of `math.pow(0.0, -math.inf)` and `math.pow(-0.0, -math.inf)` to return positive infinity instead of raising `ValueError`. This makes `math.pow` consistent with the built-in `pow` (and the `**` operator) for this particular special case, and brings the `math.pow` special-case handling into compliance with IEEE 754.

Before:
```python
Python 3.11.0a0 (heads/main:257e400a19, Jun  8 2021, 18:00:50) [Clang 11.0.0 (clang-1100.0.33.17)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import math
>>> math.pow(0.0, -math.inf)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: math domain error
>>> math.pow(-0.0, -math.inf)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: math domain error
```

After:
```python
Python 3.11.0a0 (heads/fix-float-pow-ieee-754-corner-case:1be10ea32f, Jun  8 2021, 17:43:05) [Clang 11.0.0 (clang-1100.0.33.17)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import math
>>> math.pow(0.0, -math.inf)
inf
>>> math.pow(-0.0, -math.inf)
inf
```

https://bugs.python.org/issue44339

### CODE REVIEW

File: Lib/test/ieee754.txt [hunk 1]
```diff
@@ -182,4 +180,3 @@ ValueError: math domain error
 
 Hyberbolic Functions
 ====================
-
```
Comment (mdickinson) [inline Lib/test/ieee754.txt]: Accidental change; will revert.

Comment (mdickinson) [reply Lib/test/ieee754.txt]: Done.

**Review by erlend-aasland — APPROVED** [review]

File: Lib/test/test_math.py [hunk 2]
```diff
@@ -1214,7 +1214,7 @@ def testPow(self):
         self.assertRaises(ValueError, math.pow, 0., -2.)
         self.assertRaises(ValueError, math.pow, 0., -2.3)
         self.assertRaises(ValueError, math.pow, 0., -3.)
-        self.assertRaises(ValueError, math.pow, 0., NINF)
+        self.assertEqual(math.pow(0., NINF), INF)
```
Comment (corona10) [inline Lib/test/test_math.py:1233]: ~For the consistent result check,~
~I would like to suggest adding below tests~

````python
self.assertEqual(math.pow(0., NINF), 0. ** NINF)
self.assertEqual(math.pow(-0., NINF), (-0.) ** NINF)
````

**Review by corona10 — CHANGES_REQUESTED** [review]

File: Lib/test/test_math.py (hunk 2 above)
Comment (corona10) [reply Lib/test/test_math.py:1233]: Ah, I read the following comment now :( .
https://github.com/python/cpython/blob/40a928429d7f3951c0a593f39d1b26f7e02a7529/Lib/test/test_math.py#L1330

Please ignore my comment.

**Review by corona10 — APPROVED** [review]

**Pull request merged**